src/devices/Hx711/HX711Reader.cs

```diff
@@ -0,0 +1,321 @@
+
+            // If we're only average across one value, just read it and return it.
+            if (numberOfReads == 1)
+            {
+                return CalculateNetValue(ReadInt(), offsetFromZero);
+            }
+
+            // If we're averaging across a low amount of values, just take the
+            // median.
+            if (numberOfReads < 5)
+            {
+                return CalculateNetValue(ReadMedian(numberOfReads), offsetFromZero);
+            }
+
+            return CalculateNetValue(ReadAverage(numberOfReads), offsetFromZero);
+        }
+
+        /// <summary>
+        /// Calculate net value
+        /// </summary>
+        /// <param name="value">Gross value read from HX711</param>
+        /// <param name="offset">Offset value from 0</param>
+        /// <returns>Return net value read</returns>
+        private static int CalculateNetValue(int value, int offset)
+        {
+            return value - offset;
+        }
+
+        /// <summary>
+        /// HX711 Channel and gain factor are set by number of bits read
+        /// after 24 data bits.
+        /// </summary>
+        /// <param name="mode">Current HX711 mode</param>
+        /// <returns>Number of extrabit after 24 bit</returns>
+        /// <exception cref="ArgumentOutOfRangeException">Throw if mode value is invalid.</exception>
+        /// <remarks>Look table "Table 3 Input Channel and Gain Selection" in doc page 4
+        /// https://html.alldatasheet.com/html-pdf/1132222/AVIA/HX711/457/4/HX711.html</remarks>
+        private static int CalculateExtraBitByMode(Hx711Mode mode)
+        {
+            switch (mode)
+            {
+                case Hx711Mode.ChannelAGain128:
+                    return 1;
+                case Hx711Mode.ChannelBGain32:
+                    return 2;
+                case Hx711Mode.ChannelAGain64:
+                    return 3;
+                default:
+                    throw new ArgumentOutOfRangeException(paramName: nameof(mode), message: "Unknow HX711 mode.");
+            }
+        }
+
+        /// <summary>
+        /// The output 24 bits of data is in 2's complement format. Convert it to int.
+        /// </summary>
+        /// <param name="inputValue">24 bit in 2' complement format</param>
+        /// <returns>Int converted</returns>
+        private static int ConvertFromTwosComplement24bit(int inputValue)
+        {
+            // Docs says
+            // "When input differential signal goes out of the 24-bit range,
+            // the output data will be saturated at 800000h (MIN) or 7FFFFFh (MAX),
+            // until the input signal comes back to the input range.", page 4
+            // https://html.alldatasheet.com/html-pdf/1132222/AVIA/HX711/457/4/HX711.html
+
+            // 24 bit in 2's complement only 23 are a value if
+            // the number is negative. 0xFFFFFF >> 1 = 0x7FFFFF
+            // Mask to take true value
+            const int MAX_VALUE = 0x7FFFFF;

```

Comment:
MaxValue and BitSign for the naming convention

Reply:
Ok, i fixed it